decidim-proposals/app/models/decidim/proposals/proposal.rb

```diff
@@ -55,18 +55,19 @@ class Proposal < Proposals::ApplicationRecord
 
       geocoded_by :address
 
-      scope :answered, -> { where.not(answered_at: nil) }
-      scope :not_answered, -> { where(answered_at: nil) }
-
-      scope :state_not_published, -> { where(state_published_at: nil) }
-      scope :state_published, -> { where.not(state_published_at: nil).where.not(state: nil) }
+      enum state: POSSIBLE_STATES
```

Comment:
How about renaming this one so it's consistent with the rest of STATES constants? We're using `Decidim::Amendment::STATES`, `Decidim::ReminderRecord::STATES`, `Decidim::Proposals::CollaborativeDraft::STATES` ... and `Decidim::Proposals::Proposal::POSSIBLE_STATES` 
```suggestion
      enum state: STATES
```